portal-impl/src/com/liferay/portal/servlet/FriendlyURLServlet.java

```diff
@@ -393,6 +393,11 @@ else if (_log.isWarnEnabled()) {
 			group.getGroupId(), _private, Portal.PATH_MAIN, friendlyURL, params,
 			requestContext);
 
+		if (actualURL.startsWith(PortalUtil.getPortalURL(request))) {
+			actualURL = StringUtil.removeSubstring(
+				actualURL, PortalUtil.getPortalURL(request));
+		}
+
```

Comment:
@ling-alan-huang can you help me understand how this change is related?  Your change in PortalImpl was effective by itself - does this cover another edge case, or is it a quality improvement that is not directly necessary to fix the issue?

Reply:
cc @hhuijser

Reply:
@ling-alan-huang no problem, thanks for clarifying!